core/kazoo_number_manager-1.0.0/src/knm_phone_number.erl

```diff
@@ -0,0 +1,549 @@
+    Routines = [
+        fun knm_providers:delete/1
+        ,fun delete_number_doc/1
+        ,fun maybe_remove_number_from_account/1
+    ],
+    routines(Number, Routines).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec to_public_json(number()) -> wh_json:object().
+to_public_json(Number) ->
+    JObj = to_json(Number),
+    wh_json:public_fields(JObj).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec to_json(number()) -> wh_json:object().
+to_json(#number{doc=JObj}=N) ->
+    Now = wh_util:current_tstamp(),
+    wh_json:from_list(
+        props:filter_undefined([
+            {<<"_id">>, number(N)}
+            ,{?PVT_DB_NAME, number_db(N)}
+            ,{?PVT_ASSIGNED_TO, assigned_to(N)}
+            ,{?PVT_PREVIOUSLY_ASSIGNED_TO, prev_assigned_to(N)}
+            ,{?PVT_USED_BY, used_by(N)}
+            ,{?PVT_FEATURES, features(N)}
+            ,{?PVT_STATE, state(N)}
+            ,{?PVT_RESERVE_HISTORY, reserve_history(N)}
+            ,{?PVT_PORTED_IN, ported_in(N)}
+            ,{?PVT_MODULE_NAME, module_name(N)}
+            ,{?PVT_REGION, region(N)}
+            ,{?PVT_MODIFIED, Now}
+            ,{?PVT_CREATED, wh_json:get_value(<<"pvt_created">>, JObj, Now)}
+            ,{?PVT_TYPE, <<"number">>}
+        ])
+    ).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec from_json(wh_json:object()) -> number().
+from_json(JObj) ->
+    #number{
+        number=wh_json:get_value(<<"_id">>, JObj)
+        ,number_db=wh_json:get_value(?PVT_DB_NAME, JObj)
+        ,assigned_to=wh_json:get_value(?PVT_ASSIGNED_TO, JObj)
+        ,prev_assigned_to=wh_json:get_value(?PVT_PREVIOUSLY_ASSIGNED_TO, JObj)
+        ,used_by=wh_json:get_value(?PVT_USED_BY, JObj)
+        ,features=wh_json:get_value(?PVT_FEATURES, JObj)
+        ,state=wh_json:get_value(?PVT_STATE, JObj)
+        ,reserve_history=wh_json:get_value(?PVT_RESERVE_HISTORY, JObj)
+        ,ported_in=wh_json:get_value(?PVT_PORTED_IN, JObj)
+        ,module_name=wh_json:get_value(?PVT_MODULE_NAME, JObj)
+        ,region=wh_json:get_value(?PVT_REGION, JObj)
+        ,doc=JObj
+    }.
+
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec new() -> number().
+new() ->
+    #number{}.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec setters(number(), wh_proplist()) -> number().
+setters(Number, Props) ->
```

Comment:
Similar to `routines`, this sounds like getting a property called 'setters'. Why not `set_values`?